### PR TITLE
Attempt to handle non-master git default branches

### LINKED ISSUE #1742: Handle non-master default git branches
gopass has the hardcoded expectation that the git default branch is always called `master`.
This default is changing upstream and we should support whatever users decide their default branch should be called.

### PR DESCRIPTION
**Pull request by dominikschulz**
Fixes #1742

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

File: internal/action/rcs.go [hunk 1]
```diff
@@ -136,10 +137,10 @@ func (s *Action) RCSPush(c *cli.Context) error {
 	origin := c.Args().Get(0)
 	branch := c.Args().Get(1)
 
-	if origin == "" || branch == "" {
-		return ExitError(ExitUsage, nil, "Usage: %s git push <ORIGIN> <BRANCH>", s.Name)
-	}
 	if err := s.Store.RCSPush(ctx, store, origin, branch); err != nil {
+		if errors.Is(err, si.ErrGitNoRemote) {
+			return nil
```
Comment (AnomalRoil) [inline internal/action/rcs.go:142]: We might want to log that out occurred, no?

File: internal/backend/storage/gitfs/git.go [hunk 2]
```diff
@@ -235,7 +235,7 @@ func (g *Git) defaultRemote(ctx context.Context, branch string) string {
 func (g *Git) defaultBranch(ctx context.Context) string {
 	out, _, err := g.captureCmd(ctx, "defaultBranch", "rev-parse", "--abbrev-ref", "HEAD")
 	if err != nil || string(out) == "" {
-		return "master"
+		return "main"
```
Comment (AnomalRoil) [inline internal/backend/storage/gitfs/git.go:238]: Why change the default?

**Review by AnomalRoil — APPROVED** [review]

File: internal/action/rcs.go (hunk 1 above)
Comment (dominikschulz) [reply internal/action/rcs.go:142]: Why bother if no remote is configured?
But yes, we could log this.

File: internal/backend/storage/gitfs/git.go (hunk 2 above)
Comment (dominikschulz) [reply internal/backend/storage/gitfs/git.go:238]: See e.g. https://github.com/github/renaming for context.

**Pull request merged**